engine/gamesys/src/gamesys/scripts/script_window.cpp

```diff
@@ -168,6 +170,36 @@ static int SetListener(lua_State* L)
     return 0;
 }
 
+/*# set the locking state for current mouse cursor
+ *
+ * Set the locking state for current mouse cursor on a PC platform.
+ *
+ * This function locks or unlocks the mouse cursor to the center point of the window. While the cursor is locked,
+ * mouse position updates will still be sent to the scripts as usual.
+ *
+ * @name window.set_mouse_lock
+ * @param flag [type:boolean] The lock state for the mouse cursor
+ */
+static int SetMouseLock(lua_State* L)
+{
+    int top = lua_gettop(L);
+
+    bool flag = dmScript::CheckBoolean(L, 1);
+
+    // Hiding the cursor is the same thing as locking it currently
+    if (flag)
+    {
+        dmHID::HideMouseCursor(g_Window.m_HidContext);
+    }
+    else
+    {
+        dmHID::ShowMouseCursor(g_Window.m_HidContext);
+    }
```

Comment:
Hm, maybe rename these to HideAndLock* or something like that, since that's more what they do in reality